### PR TITLE
Add circuit separation into non-Clifford and Clifford halves

### PR DESCRIPTION
**Pull request by cduck**
Final part of #462.

### CODE REVIEW

File: cirq/contrib/paulistring/separate.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    Returns:
+        A Circuit with only PauliStringPhasor operations.
+    """
+    return circuits.Circuit.from_ops(
+            _pull_non_clifford_before(circuit),
+            strategy=circuits.InsertStrategy.EARLIEST)
+
+
+def _pull_non_clifford_before(circuit: circuits.Circuit) -> ops.OP_TREE:
+    def _iter_ops_range_reversed(moment_start, moment_end):
```
Comment (Strilanc) [inline cirq/contrib/paulistring/separate.py]: constant param moment_start is always 0

File: cirq/contrib/paulistring/separate_test.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    c_left, c_right = convert_and_separate_circuit(circuit)
+
+    cirq.testing.assert_allclose_up_to_global_phase(
+        circuit.to_unitary_matrix(),
+        (c_left + c_right).to_unitary_matrix(),
+        atol=1e-7,
+    )
+
+    print(repr(c_left))
```
Comment (Strilanc) [inline cirq/contrib/paulistring/separate_test.py]: debug line

File: cirq/contrib/paulistring/separate.py (hunk 1 above)
Comment (cduck) [reply cirq/contrib/paulistring/separate.py]: Done.

File: cirq/contrib/paulistring/separate_test.py (hunk 2 above)
Comment (cduck) [reply cirq/contrib/paulistring/separate_test.py]: Done.

**Comment by cduck** [comment]
I also fixed the docstrings and removed an unneeded function.

**Pull request merged**